src/object.d

```diff
@@ -3012,6 +3015,36 @@ Key[] keys(T : Value[Key], Value, Key)(T *aa) @property
     assert(s.keys.length == 0);
 }
 
+@safe unittest
+{
+    @safe static struct Key
+    {
+         string str;
+         this(this) @safe {}
+    }
+    string[Key] aa;
+    static assert(__traits(compiles, {
+                void test() @safe {
+                    const _ = aa.keys;
+                }
+            }));
+}
+
+@safe unittest
+{
+    static struct Key
+    {
+        string str;
+        this(this) @system {}
```

Comment:
Test with @system postblit is here, @RazvanN7

Reply:
> Needed:
> 
>     * referenced bugzilla entry

Do you have any advice on how to find the right bugzilla entry?

Reply:
not really. If you don't find anything, you can just submit it yourself

Reply:
Found it, thanks.